src/resources/Search/SearchInterfaces.ts

```diff
@@ -39,6 +39,244 @@ export interface TokenModel {
     token: string;
 }
 
+export type RestQueryParams = PostSearchBodyQueryParams & PostSearchQueryStringParams;
+
 export interface PostSearchQueryStringParams {
-    viewAllContent?: boolean;
+    viewAllContent?: boolean | number;
+}
+
+export interface PostSearchBodyQueryParams {
+    actionsHistory?: RestActionHistory[];
+    analytics?: RestAnalyticsRequest;
+    aq?: string;
+    categoryFacets?: RestCategoryFacetRequest[];
+    childField?: string;
+    commerce?: RestCommerceParameters;
+    context?: RestContextRequest;
+    cq?: string;
+    debug?: boolean;
+    dictionaryFieldContext?: RestDictionaryFieldContextRequest;
+    dq?: string;
+    enableCollaborativeRating?: boolean;
+    enableDidYouMean?: boolean;
+    enableDuplicateFiltering?: boolean;
+    enableMLDidYouMean?: boolean;
+    enableQuerySyntax?: boolean;
+    excerptLength?: number;
+    facetOptions?: RestFacetOptions[];
```

Comment:
This shouldn't be an array
```suggestion
    facetOptions?: RestFacetOptions;
```

Reply:
oops...